src/Domain/DonationService.php

```diff
@@ -345,6 +345,8 @@ function () use ($donation) {
                         // would be reduced again when the donation expires.
                         $this->donationRepository->removeAllFundingWithdrawalsForDonation($donation);
 
+                        $this->entityManager->flush();
```

Comment:
looks like this might be worth trying but but in most cases if an exception has been thrown during allocateMatchFunds it may mean the EM is closed and we won't be able to flush anyway. Does that seem right to you @NoelLH ?

Reply:
Yes, certainly sounds possible. There's a chance the EM retry methods work to sort this out in some cases, but I don't think anything has hit this case recently enough for me to be confident either way.